File: atv14/exerc14.py
class Configuracao:
    _instancia = None

    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super(Configuracao, cls).__new__(cls)
        return cls._instancia

    def exibir_configuracao(self):
        print("Exibindo a configuração...")

config1 = Configuracao()
config2 = Configuracao()

config1.exibir_configuracao()
print(config1 is config2)  # True, pois ambas são a mesma instância

File: atv14/exerc14.go
package main
import (
    "fmt"
    "sync"
)

type Configuracao struct{}

var instancia *Configuracao
var once sync.Once

func GetInstancia() *Configuracao {
    once.Do(func() {
        instancia = &Configuracao{}
    })
    return instancia
}

func (c *Configuracao) ExibirConfiguracao() {
    fmt.Println("Exibindo a configuração...")
}

func main() {
    config1 := GetInstancia()
    config2 := GetInstancia()

    config1.ExibirConfiguracao()
    fmt.Println(config1 == config2)  // true, porque ambas são a mesma instância
}
